### PR TITLE
add support for elasticsearch reporter settings

### PR DESCRIPTION
**Pull request by oleksiyp**
**Description**

Adding possibility to specify ElasticSearch index settings

### CODE REVIEW

**Comment by gaetanmaisse** [comment]
Thanks for the contribution @oleksiyp 💪🏻  we will review it in the next couple of days!

File: apim/3.x/templates/gateway/gateway-configmap.yaml [hunk 1]
```diff
@@ -216,6 +216,10 @@ data:
             health: {{ .Values.es.lifecycle.policies.health }}
             log: {{ .Values.es.lifecycle.policies.log }}
         {{- end }}
+        settings:
```
Comment (gaetanmaisse) [inline apim/3.x/templates/gateway/gateway-configmap.yaml:219]: I think we need to add a `{{ if  .Values.es.settings}}` around the whole code you added, to be sure everything is still running fine when `number_of_shards`, `number_of_replicas` and `refresh_interval` are not explicitly defined.

**Review by gaetanmaisse — COMMENTED** [review]
Also, can you add some unit tests to cover this part? 
A super small doc is available here: https://github.com/gravitee-io/helm-charts/blob/master/apim/3.x/TESTS.md
And examples there: https://github.com/gravitee-io/helm-charts/tree/master/apim/3.x/tests/gateway

**Comment by sebdevaux** [comment]
Closed in favor of https://github.com/gravitee-io/gravitee-api-management/pull/4904
Thank you @oleksiyp

**Pull request closed without merging**